core/src/App.html

```diff
@@ -1045,12 +1045,25 @@
     internalUserSettingsObject.displayed = false;
   };
 
+  // Open View in New Tab
+
+  const openViewInNewTab = async nodepath => {
+    const { nodeObject } = await Navigation.extractDataFromPath(nodepath);
+    if (await Navigation.shouldPreventNavigation(nodeObject)) {
+      return;
+    }
```

Comment:
It seems this part is repeated as function `openViewInDrawer` and `openViewInModal`. Are we able to extract to a single function?

Reply:
I notice that 🤔 . I am trying to refactor it though, but it is so far seeming like we don't end up saving a lot of lines 😄 . If you compare **old usage** vs. **new usage** below seems like there's only one single line difference:
```
    // old usage
    const { nodeObject } = await Navigation.extractDataFromPath(nodepath);
    if (await Navigation.shouldPreventNavigation(nodeObject)) {
      return;
    }
```

```
// new function
shouldPreventNavigation(nodepath) {
    const { nodeObject } = await Navigation.extractDataFromPath(nodepath);
    if (await Navigation.shouldPreventNavigation(nodeObject)) {
      return true;
    }
    return false;
}

// new usage
if (shouldPreventNavigation(nodepath)) {
      return;
}
```
Not sure if it is worth the change. Maybe you have a better suggestion than my snippet above, that would be nice. Otherwise I would leave it as is. What do you think ?

Reply:
Cool. We still definitely should go for the new approach. Saving lines is the bonus, not the first priority in this case.
The point is to put this logic in a single place. The benefit will be seen whenever we need to update the business logic.

Reply:
Added changes 👍